test/gtest/binary_tensor_ops.cpp

```diff
@@ -0,0 +1,295 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <gtest/gtest.h>
+
+#include <miopen/tensor.hpp>
+#include <miopen/tensor_ops.hpp>
+
+#include "get_handle.hpp"
+#include "tensor_holder.hpp"
+#include "tensor_util.hpp"
+#include "verify.hpp"
+
+namespace {
+using BinaryTensorOpsCase = std::tuple<std::vector<size_t>, std::vector<int>, float, bool>;
+
+template <typename DstType, typename SrcType = DstType>
+class GPU_binaryTensorOps : public ::testing::TestWithParam<BinaryTensorOpsCase>
+{
+public:
+    static tensor<DstType> dstSuperTensor;
+    static tensor<SrcType> srcSuperTensor;
+
+    static void SetUpTestSuite()
+    {
+        static constexpr auto dstType = miopen_type<DstType>{};
+        static constexpr auto srcType = miopen_type<SrcType>{};
+
+        uint64_t dstMaxValue = dstType == miopenHalf ? 5 : (dstType == miopenInt8 ? 126 : 32767);
+        uint64_t srcMaxValue = srcType == miopenHalf ? 5 : (srcType == miopenInt8 ? 126 : 32767);
+
+        uint64_t maxValue = std::min(dstMaxValue, srcMaxValue);
```

Comment:
For the old tests, it looks like they had different max values being used depending on the mode.

Here are the ones for tensor_copy.cpp for example:
```cpp
               uint64_t max_value = miopen_type<T>{} == miopenHalf   ? 5
                             : miopen_type<T>{} == miopenInt8 ? 127
                                                              : 17;
```

I think it would make sense to keep the max values the same as before for the different tests. 
That being said, it also doesn't seem like it's a huge difference, and is probably okay without it.
Sort of curious why they were like that before.

Reply:
The old test did `int` -> `dstType` conversion, so it was focused on dst type only.
Now it is `srcType` -> `dstType` conversion and we must fit into both.
If you are care about `17` vs `32768` - for me if does make sense because the old values were just random numbers and for the operations themselves it doesn't matter, especially for `Copy`. I took it from the `Cast` which is kind of sensitive to the input data and I see not point to change it for `Copy` which can take actually any random garbage.

Reply:
Yea I agree, and think it's fine to leave it as is.